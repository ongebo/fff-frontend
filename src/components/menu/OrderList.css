button:disabled,
button:disabled:hover {
  background: lightgray;
}
