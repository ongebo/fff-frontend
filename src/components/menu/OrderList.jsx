import React, { Component } from 'react';
import { connect } from 'react-redux';
import { placeOrder } from '../../actions/FoodOrderActions';
import './OrderList.css';

export class OrderList extends Component {
  constructor(props) {
    super(props);
    this.order = this.order.bind(this);
  }

  order() {
    const orderData = {
      items: this.props.orderList.map(item => ({
        item: item.item,
        quantity: item.quantity,
        cost: item.cost
      }))
    };
    this.props.placeOrder(this.props.token, orderData);
    this.props.resetOrderList();
  }

  render() {
    return (
      <div className="column side">
        <div className="order-list">
          <h2 className="title">Order List</h2>
          {this.props.orderList.length === 0 ? (
            <h2 id="empty-list">Order List is Empty!</h2>
          ) : (
            this.props.orderList.map(item => (
              <p key={item.item} className="order-list-item">{`${
                item.quantity
              } ${item.unit}s of ${item.item} @ Ugx ${item.cost}`}</p>
            ))
          )}
          <button
            disabled={this.props.orderList.length === 0 ? true : false}
            onClick={this.order}
            className="order"
          >
            Place Order
          </button>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    token: state.authentication.token
  };
}

export default connect(
  mapStateToProps,
  { placeOrder }
)(OrderList);
