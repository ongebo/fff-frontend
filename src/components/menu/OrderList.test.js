import React from 'react';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { OrderList } from './OrderList';

Enzyme.configure({ adapter: new EnzymeAdapter() });
const props = {
  orderList: [],
  resetOrderList: jest.fn(),
  placeOrder: jest.fn()
}

describe('<OrderList />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <OrderList {...props} />
    );
    wrapper.instance().order();
    expect(wrapper).toMatchSnapshot();
  });

  it('contains an order-list <div />', () => {
    const wrapper = shallow(
      <OrderList {...props} />
    );
    expect(wrapper.find('.order-list')).toHaveLength(1);
  });
});
