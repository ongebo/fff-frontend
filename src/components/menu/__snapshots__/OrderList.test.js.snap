// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderList /> matches snapshot 1`] = `ShallowWrapper {}`;
